app/Http/Controllers/Api/RoleController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\Api\RolePermissionRequest;
+use App\Http\Requests\Api\RoleRequest;
+use Spatie\Permission\Models\Role;
+
+class RoleController extends Controller
+{
+    public function index()
+    {
+        $roles = Role::all();
+
+        return response($roles);
+    }
+
+    public function store(RoleRequest $request)
+    {
+        $validated = $request->validated();
+        $role = Role::create($validated);
+
+        return response($role);
+    }
+
+    public function show(Role $role)
+    {
+        return $role;
+    }
+
+    public function update(RoleRequest $request, Role $role)
+    {
+        $validated = $request->validated();
+        $role->update($validated);
+
+        return response($role);
+    }
+
+    public function destroy(Role $role)
+    {
+        $role->delete();
+
+        return response()->noContent();
+    }
+
+    public function updateRolePermissions(RolePermissionRequest $request, Role $role)
+    {
+        $validated = $request->validated();
+        $permissions = array_pluck($validated['permissions'], 'id');
```

Comment:
what exactly are you sending in the payload for permissions? I was thinking it will be an array of ids. If yes, then why do you need to do `array_pluck`?

Reply:
Earlier I was sending the multiple objects of the `id` and `name` in the permissions array in payload that's why I used the `array_pluck` for retrieves all of the values for a given key from an array. So the preveious payload structure is something like this,
```
"permissions": [
    { "name": "clients.create", "id": "1" },
    { "name": "clients.view", "id": "2" }
]
```
but now I thought that when the permission name is not used then we can directly send only permission ids in the payload and update directly. Can according to this I changed the code and now I am sending the array of ids in the payload and updating directly.
Now, I send the `permissions` in the payload, something like this.
```
"permissions": [
    1,
    2
]
```